frontend/wallet/src/render/components/RequestCodaModal.re

```diff
@@ -0,0 +1,98 @@
+open Tc;
+
+module Styles = {
+    open Css;
+  
+    let bodyMargin = style([
+      margin(`rem(1.0)),
+      width(`percent(100.))
+    ]);
+
+    let publicKey = merge([
+      Theme.Text.Body.regular,
+      style([
+        border(`px(1), `solid, Theme.Colors.marineAlpha(0.3)),
+        borderRadius(`rem(0.25)),
+        padding(`px(12)),
+        backgroundColor(white),
+        wordWrap(`breakWord),
+      ]),
+    ]);
+  };
+    
+  [@react.component]
+  let make = (~wallets, ~setModalState) => {
+    let activePublicKey = Hooks.useActiveWallet();
+    let (settings, _updateAddressBook) = React.useContext(AddressBookProvider.context);
+    let (selectedWallet, setSelectedWallet) = React.useState(() => activePublicKey);
+    let keyString = switch (selectedWallet) {
+      | None => ""
```

Comment:
This empty string state never gets used, as far as I can tell. You could probably avoid needing to represent this state by only using it in your switch cases below (instead of matching against `Some(_selectedWallet)`)